lua/sim/Projectile.lua

```diff
@@ -141,20 +141,26 @@ Projectile = ClassProjectile(ProjectileMethods) {
     end,
 
     --- Called by Lua during the `OnCreate` event when the blueprint field `TrackTargetGround` is set,
-    --- used by tactical missiles to track a patch of ground in the vicinity of the unit
+    --- used by tactical missiles to track a point around/inside the target's hitbox
     ---@param self Projectile
     OnTrackTargetGround = function(self)
         local target = self.OriginalTarget or self:GetTrackingTarget() or self.Launcher:GetTargetEntity()
         if target and target.IsUnit then
+
             local unitBlueprint = target.Blueprint
-            local cy = unitBlueprint.CollisionOffsetY or 0
+
+            -- X-offset units often have displaced center bones, so they're not accounted for.
+            local cy, cz = unitBlueprint.CollisionOffsetY or 0, unitBlueprint.CollisionOffsetZ or 0
             local sx, sy, sz = unitBlueprint.SizeX or 1, unitBlueprint.SizeY or 1, unitBlueprint.SizeZ or 1
             local px, py, pz = target:GetPositionXYZ()
 
-            -- take into account heading
-            local heading = -1 * target:GetHeading() -- inverse heading because Supreme Commander :)
-            local mch = MathCos(heading)
-            local msh = MathSin(heading)
+            -- add offset to prefer the lower end of the hitbox
+            if cy == 0 then
+                cy = sy / 4
+            elseif cy * 2 < sy / 2 then
+                sy = sy - (0.5 * sy + 2 * cy)
+                cy = sy / 2
+            end
```

Comment:
I noticed the subtle preference here to target the lower end of a unit, is that there to fix a bug or to increase the accuracy of missiles @lL1l1 😃

Reply:
It's not meant to bias the offset in any direction, only change the targeting volume to the above-surface part of the hitbox.
I think I made a mistake with the parentheses. It should be `sy = sy - 0.5 * sy + 2 * cy`
![image](https://github.com/FAForever/fa/assets/82986251/e9342347-b156-4166-826b-81681c6d32c1)
Also there's a mistake in `cy = sy / 2` where it should be `cy = sy / 4`

Reply:
> It's not meant to bias the offset in any direction, only change the targeting volume to the above-surface part of the hitbox. The comments in the original code explained this.

I don't know how I made the mistake of thinking the offset is 2x or acted so weirdly.
```suggestion
            -- don't target the part of the hitbox below the surface
            if cy < 0 then
                sy = sy + cy
                cy = 0
            end
```

Reply:
Unexpectedly megalith changes its collision box when it finishes building to values different from its blueprint. That's what I was compensating for in my testing process when writing the code.